doc/api/url.md

```diff
@@ -734,346 +405,681 @@ to customize the serialization process of the URL. For more flexibility,
+  Defaults to `false`.
 
-Returns an ES6 Iterator over the values of each name-value pair.
+The `url.parse()` method takes a URL string, parses it, and returns a URL
+object.
 
-#### urlSearchParams\[@@iterator\]()
+A `TypeError` is thrown if `urlString` is not a string.
 
-* Returns: {Iterator}
+A `URIError` is thrown if the `auth` property is present but cannot be decoded.
 
-Returns an ES6 Iterator over each of the name-value pairs in the query string.
-Each item of the iterator is a JavaScript Array. The first item of the Array
-is the `name`, the second item of the Array is the `value`.
+### url.resolve(from, to)
+<!-- YAML
+added: v0.1.25
+changes:
+  - version: v6.6.0
+    pr-url: https://github.com/nodejs/node/pull/8215
+    description: The `auth` fields are now kept intact when `from` and `to`
+                 refer to the same host.
+  - version: v6.5.0, v4.6.2
+    pr-url: https://github.com/nodejs/node/pull/8214
+    description: The `port` field is copied correctly now.
+  - version: v6.0.0
+    pr-url: https://github.com/nodejs/node/pull/1480
+    description: The `auth` fields is cleared now the `to` parameter
+                 contains a hostname.
+-->
 
-Alias for [`urlSearchParams.entries()`][].
+* `from` {string} The Base URL being resolved against.
+* `to` {string} The HREF URL being resolved.
+
+The `url.resolve()` method resolves a target URL relative to a base URL in a
+manner similar to that of a Web browser resolving an anchor tag HREF.
+
+For example:
 
 ```js
-const { URLSearchParams } = require('url');
-const params = new URLSearchParams('foo=bar&xyz=baz');
-for (const [name, value] of params) {
-  console.log(name, value);
-}
-  // Prints:
-  // foo bar
-  // xyz baz
+url.resolve('/one/two/three', 'four');         // '/one/two/four'
+url.resolve('http://example.com/', '/one');    // 'http://example.com/one'
+url.resolve('http://example.com/one', '/two'); // 'http://example.com/two'
 ```
 
-<a id="whatwg-percent-encoding"></a>
-### Percent-Encoding in the WHATWG URL Standard
+## Percent-Encoding in URLs
 
 URLs are permitted to only contain a certain range of characters. Any character
 falling outside of that range must be encoded. How such characters are encoded,
 and which characters to encode depends entirely on where the character is
-located within the structure of the URL. The WHATWG URL Standard uses a more
-selective and fine grained approach to selecting encoded characters than that
-used by the older [`url.parse()`][] and [`url.format()`][] methods.
+located within the structure of the URL.
+
+### Legacy API
+
+Within the Legacy API, spaces (`' '`) and the following characters will be
+automatically escaped in the properties of URL objects:
+
+```txt
+< > " ` \r \n \t { } | \ ^ '
+```
+
+For example, the ASCII space character (`' '`) is encoded as `%20`. The ASCII
+forward slash (`/`) character is encoded as `%3C`.
+
+<a id="whatwg-percent-encoding"></a>
```

Comment:
It is moved since James refactored the original chapter "Percent-Encoding in WHATWG" to a chapter "Percent-Encoding" with two sections "Legacy" and "WHATWG". The anchor goes with the description of WHATWG URL.